### ISSUE TITLE
Implement incremental update using commit stats (SnapshotSummary)

### ISSUE THREAD

**Issue by ajantha-bhat** [apache/iceberg#8461]
### Feature Request / Improvement

This could be an experimental direction and can be controlled by a flag.

This might bloat up the table metadata file size when millions of partitions are present. 

### Query engine

None

**Comment by ajantha-bhat** [comment]
Based on the experiments from https://github.com/apache/iceberg/pull/9437, spark action is not effective as the serialization cost of each partition stats entry is expensive. 

Will implement an API in core module to compute stats in a distributed way and a spark procedure to call this API.

**Comment by ajantha-bhat** [comment]
Lower priority and will be working on this once all other planned activities from #8450 is done.
